src/components/app/details/cdDetails/DeploymentHistoryConfigTabView.tsx

```diff
@@ -1,87 +1,70 @@
-    const [baseTemplateConfiguration, setBaseTemplateConfiguration] = useState<DeploymentTemplateViaTargetId>();
+    const { appId, pipelineId } = useParams<{ appId: string; pipelineId: string }>()
+    const [selectedDeploymentTemplate, setSeletedDeploymentTemplate] = useState<DeploymentTemplateOptions>()
+    const [currentConfiguration, setCurrentConfiguration] = useState<DeploymentTemplateViaTargetId>()
+    const [baseTemplateConfiguration, setBaseTemplateConfiguration] = useState<DeploymentTemplateViaTargetId>()
 
-    const [loader, setLoader] = useState<boolean>(false);
-    const [codeEditorLoading, setCodeEditorLoading] = useState<boolean>(false);
+    const [codeEditorLoading, setCodeEditorLoading] = useState<boolean>(false)
 
     useEffect(() => {
-        setLoader(true);
-        if (selectedDeploymentTemplate) {
+        setLoader(true)
+
+        if (deploymentTemplatesConfiguration && selectedDeploymentTemplate) {
             try {
                 getDeploymentTemplateDiffId(appId, pipelineId, selectedDeploymentTemplate.value).then((response) => {
-                    setCurrentConfiguration(response.result);
-                    setLoader(false);
-                });
+                    setCurrentConfiguration(response?.result)
+                    setLoader(false)
+                })
             } catch (err) {
-                showError(err);
-                setLoader(false);
+                showError(err)
+                setLoader(false)
             }
         }
-    }, [selectedDeploymentTemplate]);
+    }, [selectedDeploymentTemplate])
 
     useEffect(() => {
         try {
-            setCodeEditorLoading(true);
-            if (baseTemplateId) {
+            setCodeEditorLoading(true)
+            if (deploymentTemplatesConfiguration && baseTemplateId) {
                 getDeploymentTemplateDiffId(appId, pipelineId, baseTemplateId).then((response) => {
-                    setBaseTemplateConfiguration(response.result);
-                    setCodeEditorLoading(false);
-                });
+                    setBaseTemplateConfiguration(response.result)
+                    setCodeEditorLoading(false)
+                })
             }
         } catch (err) {
-            showError(err);
-            setCodeEditorLoading(false);
+            showError(err)
+            setCodeEditorLoading(false)
         }
-    }, [baseTemplateId]);
+    }, [baseTemplateId])
 
     useEffect(() => {
-        setLoader(true);
-        try {
-            getDeploymentTemplateDiff(appId, pipelineId).then((response) => {
-                setDeploymentTemplatesConfiguration(response.result.sort((a, b) => sortCallback('id', b, a)));
-                setLoader(false);
-            });
-
-            if (!showTemplate) {
-                setShowTemplate(true);
-            }
-        } catch (err) {
-            showError(err);
-            setLoader(false);
+        if (!showTemplate) {
+            setShowTemplate(true)
```

Comment:
Suggestion - we can set showTemplate as false on unmount if not doing explicitly.